### PR TITLE
Implement unlisted users

### PR DESCRIPTION
**Pull request by Ninjaclasher**
Finishes #539.

### CODE REVIEW

File: judge/migrations/0080_unlisted_users.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Generated by Django 1.11.13 on 2019-01-05 22:12
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('judge', '0079_remove_comment_title'),
```
Comment (kiritofeng) [inline judge/migrations/0080_unlisted_users.py]: You'll have to update this after 5f07dabd07e8fe43537b3a9f74eba7a12de8580a.

File: judge/models/problem.py [hunk 2]
```diff
@@ -247,9 +247,9 @@ def clarifications(self):
         return ProblemClarification.objects.filter(problem=self)
 
     def update_stats(self):
-        self.user_count = self.submission_set.filter(points__gte=self.points, result='AC').values('user').distinct().count()
+        self.user_count = self.submission_set.filter(points__gte=self.points, result='AC', user__is_unlisted=False).values('user').distinct().count()
```
Comment (kiritofeng) [inline judge/models/problem.py]: You should have your migration recompute this value.

File: judge/models/problem.py [hunk 3]
```diff
@@ -247,9 +247,9 @@ def clarifications(self):
...
         submissions = self.submission_set.count()
-        self.ac_rate = 100.0 * self.submission_set.filter(points__gte=self.points, result='AC').count() / submissions if submissions else 0
+        self.ac_rate = 100.0 * self.submission_set.filter(points__gte=self.points, result='AC', user__is_unlisted=False).count() / submissions if submissions else 0
```
Comment (kiritofeng) [inline judge/models/problem.py]: You should have your migration recompute this value.

File: judge/migrations/0081_unlisted_users.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# Generated by Django 1.11.13 on 2019-01-05 22:12
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('judge', '0080_contest_banned_users'),
```
Comment (Xyene) [inline judge/migrations/0081_unlisted_users.py]: Needs rebasing.

**Pull request merged**